src/DurableTask.Core/TaskOrchestrationDispatcher.cs

```diff
@@ -697,6 +697,11 @@ bool ReconcileMessagesWithState(TaskOrchestrationWorkItem workItem)
                     continue;
                 }
 
+                if (Activity.Current != null && message.Event is TimerFiredEvent)
```

Comment:
You can avoid the cast on line 702 by declaring a variable in your `is` expression.
```suggestion
                if (Activity.Current != null && message.Event is TimerFiredEvent timerFired)
```